anaconda.py

```diff
@@ -577,9 +577,26 @@ def _earlyExceptionHandler(ty, value, traceback):
     blivet.enable_installer_mode()
 
     # Initialize the network now, in case the display needs it
-    from pyanaconda.network import networkInitialize, wait_for_connecting_NM_thread
+    from pyanaconda.network import networkInitialize, wait_for_connecting_NM_thread, wait_for_connected_NM
 
     networkInitialize(ksdata)
+    # If required by user, wait for connection before starting the installation.
+    if opts.waitfornet:
+        try:
+            timeout = int(opts.waitfornet)
+        except ValueError as e:
+            log.warning("invalid inst.waitfornet option value: %s",
+                         opts.waitfornet)
+        else:
+            if timeout > 0:
+                log.info("network: waiting for network connection (timeout=%d)",
+                         timeout)
+                wait_for_connected_NM(timeout=timeout)
+            else:
+                log.warning("invalid inst.waitfornet option value: %s",
+                            opts.waitfornet)
+
+    # In any case do some actions only after NM finishes its connecting.
```

Comment:
Could you please extract this to the outside method. I don't want to feed this Godzilla method more than it is required.